ibis/common/bases.py

```diff
@@ -13,20 +13,43 @@
     from typing_extensions import Self
 
 
-class BaseMeta(type):
+class AbstractMeta(type):
```

Comment:
There wasn't much advantage of having a `Base` and an `Abstract(Base)` since the `__abstractmethod__` calculation should be cheap because it is happening on class definition.

Reply:
Pulled out to a separate PR.